### PR TITLE
Turned SDK APIs into context manager

### LINKED ISSUE #145: Enhancement: implement context manager decorator to the SDK classes
This would allow to scope the objects using the `with` statement.
Ref: https://realpython.com/python-with-statement/

### PR DESCRIPTION
**Pull request by aadityasinha-dotcom**
**What It Does**

Resolves #145 

**How to Test**

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

File: src/core/zowe/core_for_zowe_sdk/sdk_api.py [hunk 1]
```diff
@@ -56,6 +56,12 @@ def __init__(self, profile, default_url, logger_name = __name__):
             self.default_headers["Authorization"] = f"Bearer {self.session.tokenValue}"
         elif self.session.type == session_constants.AUTH_TYPE_TOKEN:
             self.default_headers["Cookie"] = f"{self.session.tokenType}={self.session.tokenValue}"
+    
+    def __enter__(self):
+        return self
+
+    def __exit__(self):
```
Comment (t1m0thyj) [inline src/core/zowe/core_for_zowe_sdk/sdk_api.py:63]: The `__exit__` method needs to take 3 additional parameters (see [PEP 343](https://peps.python.org/pep-0343/) for more details):
```suggestion
    def __exit__(self, exc_type, exception, traceback):
```

**Review by t1m0thyj — CHANGES_REQUESTED** [review]
Thanks @aadityasinha-dotcom, left a comment 🙂 

Could you please also update the examples in SDK readmes to use `with` statement? For example, in the [z/OS Console readme](https://github.com/zowe/zowe-client-python-sdk/tree/main/src/zos_console#readme):
```
from zowe.core_for_zowe_sdk import ProfileManager
from zowe.zos_console_for_zowe_sdk import Console

profile = ProfileManager().load(profile_name="zosmf")

with Console(profile) as console_info:
    print(console_info.issue_command(command="D IPLINFO", console="EMCS"))
```

File: src/core/zowe/core_for_zowe_sdk/sdk_api.py (hunk 1 above)
Comment (aadityasinha-dotcom) [reply src/core/zowe/core_for_zowe_sdk/sdk_api.py:63]: Yes you are right forgot to add that :sweat_smile:

File: src/core/zowe/core_for_zowe_sdk/sdk_api.py [hunk 2]
```diff
@@ -56,6 +56,12 @@ def __init__(self, profile, default_url, logger_name = __name__):
             self.default_headers["Authorization"] = f"Bearer {self.session.tokenValue}"
         elif self.session.type == session_constants.AUTH_TYPE_TOKEN:
             self.default_headers["Cookie"] = f"{self.session.tokenType}={self.session.tokenValue}"
+    
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exception, traceback):
```
Comment (aadityasinha-dotcom) [inline src/core/zowe/core_for_zowe_sdk/sdk_api.py:63]: Will there be any use of `exc_type, exception, traceback`? 
I am curious :thinking:

**Review by traeok — APPROVED** [review]
LGTM, thanks @aadityasinha-dotcom!

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @aadityasinha-dotcom!

**Pull request merged**